pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -941,13 +941,15 @@ ClientConnection::newConsumerStats(uint64_t consumerId, uint64_t requestId) {
 void ClientConnection::newTopicLookup(const std::string& destinationName, bool authoritative,
                                       const uint64_t requestId,
                                       LookupDataResultPromisePtr promise) {
+    Lock lock(mutex_);
```

Comment:
I mean, it's thread safe only if we do it all the time in the io thread for the particular connection :)